### PR TITLE
Introduce LoggerDiagnosticsService for improved NuGetGallery.Core logging

### PR DESCRIPTION
**Pull request by joelverhagen**
I added some tracing to NuGetGallery.Core. Tracing in NuGetGallery repository uses `IDiagnosticsService`. In NuGetGallery process this dumps to `Trace`. In orchestrator, I will write to a `ILogger`.

The traces I add provide better resolution on how a server-side blob storage copy ends up (e.g. did it de-dupe? did it replace a failed copy?).

### CODE REVIEW

**Comment by joelverhagen** [comment]
Verified in DEV with orchestrator. Also made sure E&V still launches (proper DI).

File: src/Validation.Common.Job/LoggerDiagnosticsSource.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        private static IDictionary<string, string> GetProperties(
+            IEnumerable<KeyValuePair<string, object>> payload)
+        {
+            if (payload == null)
+            {
+                return null;
+            }
+
+            var dictionary = new Dictionary<string, string>();
```
Comment (loic-sharma) [inline src/Validation.Common.Job/LoggerDiagnosticsSource.cs]: Nit: you can use LINQ's `ToDictionary` here, right? Or are you worried about key collision?

Comment (joelverhagen) [reply src/Validation.Common.Job/LoggerDiagnosticsSource.cs]: Yes, avoiding key collision.

File: src/Validation.Common.Job/LoggerDiagnosticsSource.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                dictionary[pair.Key] = pair.Value?.ToString();
+            }
+
+            return dictionary;
+        }
+
+        /// <summary>
+        /// The parameters provided to <see cref="TraceEvent(TraceEventType, int, string, string, string, int)"/>. This
+        /// type implements <see cref="IEnumerable{KeyValuePair{string, object}}" /> because Serilog uses this to
+        /// populate the Application Insights custom dimenstions property bag.
```
Comment (loic-sharma) [inline src/Validation.Common.Job/LoggerDiagnosticsSource.cs]: `dimenstions` - >`dimensions`

Comment (joelverhagen) [reply src/Validation.Common.Job/LoggerDiagnosticsSource.cs]: Fixed.

**Pull request merged**